### ISSUE TITLE
VK app authorization failed. 

### ISSUE THREAD

**Issue by trandinhlinh94** [SwiftyVK/SwiftyVK#127]
# 🚫 **ISSUE TEMPLATE** 🚫

🚫 Thanks for using SwiftyVK.

🚫 Support for this project takes a lot of my time. 

🚫 Please, be attentive, fill template and delete this title to let me cope with the issue faster. 

🚫 Issues without filled description, environment and checklist ***will be closed without any comments***

🚫 Thank you for understanding.

---

### Description

Describe the issue in detail here

### Environment

 Key                     | Value                               |
-------------------------|-------------------------------------|
 Platform                | iOS
 Platform version        | 9.0
 Xcode version           | 9.1
 Swift version           | 4.0
 SwiftyVK version        | 3.1.4
 Integration Method      | cocoapods
 
  ### Checklist
 
 Check all points of list below and set [x] on each checked point:
 
* [x ] I have looked and not find decision at [README.md](./README.md)
* [x] I have looked and reproduce this behaviour in [Example project](./Example)
* [x] I have not find duplicates of this issue in [issues list](https://github.com/west0r/SwiftyVK/issues) 
* [x] I made an understandable short title and exhaustive description
* [x ] I attach stack traces or errors description
* [ ] I completely describe the environment 
* [ ] I have full filled out this issue template

Hi, i am trying to login by VK app but when window switches to VK app, after i choose to give permission the app crashes. The reason is that it cannot find the a Web view controller which i do not know what it is. Supposed that it about web authentication. but the thing is that i use VK app the authorise user. 
It failed at this point with the fatal error. 

private func viewController<ControllerType>(
        name: String,
        onDismiss: (() -> ())?
        ) -> ControllerType {
        var controller: VKViewController?
        
        #if os(iOS)
            controller = storyboard().instantiateViewController(
                withIdentifier: name
                )
        #elseif os(macOS)
            controller = storyboard().instantiateController(
                withIdentifier: NSStoryboard.SceneIdentifier(rawValue: name)
                ) as? VKViewController
        #endif
        
        guard
            let unwrappedController = controller,
            let resultController = unwrappedController as? ControllerType
            else {
            fatalError("Can't find \(name) controller")
        }
        
        if let dismisableController = resultController as? DismisableController {
            dismisableController.onDismiss = onDismiss
        }
        
        DispatchQueue.anywayOnMain {
            self.delegate?.vkNeedToPresent(viewController: unwrappedController)
        }
        
        return resultController
    }

Please help, it's kind of urgent! Many thanks in advance!

**Comment by west0r** [comment]
Hello.
I try to reproduce this issue on:

iOS 9.3.5 and 11.2.6, 
Xcode 9.2
Cocoapods 1.2.1
And latest SwiftyVK release

but it is not reproduced.

Please, attach demo project with described error.

**Comment by trandinhlinh94** [comment]
here is a demo VC that i use to login with VK : 



    class DemoViewController: UIViewController, SwiftyVKDelegate {
    
    let scopes: Scopes = [.messages,.offline,.friends,.wall,.photos,.audio,.video,.docs,.market,.email]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        VK.setUp(appId: "6****2", delegate: self)
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func connectToVK(_ sender: Any) {
        VK.sessions.default.logIn(onSuccess: { (loginSuccess) in
            print(loginSuccess)
        }) { (error) in
            print("VK login error ", error)
        }    
    }
    
    func vkNeedToPresent(viewController: VKViewController) {
        print("present viewcontroller")
    }

    func vkNeedsScopes(for sessionId: String) -> Scopes {
        return scopes
    }
    
    func vkTokenCreated(for sessionId: String, info: [String : String]) {
        print(info)
    }
    
    func vkTokenUpdated(for sessionId: String, info: [String : String]) {
        print(info)
    }
}

AppDelegate looks like this : 

         func application(_ application: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey : Any]) -> Bool {
        SDKApplicationDelegate.shared.application(application, open: url, options: options)
        let app = options[.sourceApplication] as? String
        VK.handle(url: url, sourceApplication: app)
        return true
    }

    func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool {
        SDKApplicationDelegate.shared.application(application, open: url, sourceApplication: sourceApplication, annotation: annotation)
        VK.handle(url: url, sourceApplication: sourceApplication)
        return true
    }

I also have Fb login so that is why app delegate looks like this. I have set URL schemas and such things correctly in the Info.plist. What would you think that i did wrong here?

**Comment by west0r** [comment]
Code looks corectly, but i need demo xcodeproj with error reproducing.

You reproduce this error on Demo project  from this repo?

**Comment by trandinhlinh94** [comment]
No, i implemented this on my project. The error was coming from my project so not from a demo project of this repo. I will try to create a demo project which can reproduce the bug for you.

**Comment by west0r** [comment]
@trandinhlinh94 if you resolve this issue, please provide your solution here)

**Comment by trandinhlinh94** [comment]
Unfortunately i did not find a reason and solution for my issue. I actually have to go back to use original Obj-c lib. :( Sorry and thank you for your time!

**Comment by west0r** [comment]
See #128
